processing/src/test/java/org/apache/druid/tasklogs/SwitchingTaskLogsTest.java

```diff
@@ -0,0 +1,211 @@
+
+    verifyAll();
+  }
+
+  @Test
+  public void test_streamTaskStatus_shouldUseReportTaskLogs() throws IOException
+  {
+    String taskId = "test-task-id";
+    InputStream statusStream = new ByteArrayInputStream("test status content".getBytes(StandardCharsets.UTF_8));
+
+    EasyMock.expect(reportTaskLogs.streamTaskStatus(taskId)).andReturn(Optional.of(statusStream));
+    replayAll();
+
+    Optional<InputStream> actualStatusStream = taskLogs.streamTaskStatus(taskId);
+    Assert.assertTrue(actualStatusStream.isPresent());
+    Assert.assertEquals(statusStream, actualStatusStream.get());
+
+    verifyAll();
+  }
+
+  @Test
+  public void test_streamTaskPayload_shouldUseReportTaskLogs() throws IOException
+  {
+    String taskId = "test-task-id";
+    InputStream payloadStream = new ByteArrayInputStream("test payload content".getBytes(StandardCharsets.UTF_8));
+
+    EasyMock.expect(reportTaskLogs.streamTaskPayload(taskId)).andReturn(Optional.of(payloadStream));
+    replayAll();
+
+    Optional<InputStream> actualPayloadStream = taskLogs.streamTaskPayload(taskId);
+    Assert.assertTrue(actualPayloadStream.isPresent());
+    Assert.assertEquals(payloadStream, actualPayloadStream.get());
+
+    verifyAll();
+  }
+
+  @Test
+  public void test_pushTaskLog_shouldUsePusherTaskLogs() throws IOException
+  {
+    String taskId = "test-task-id";
+    File logFile = new File("test.log");
+
+    pusherTaskLogs.pushTaskLog(taskId, logFile);
+    EasyMock.expectLastCall();
+    replayAll();
+
+    taskLogs.pushTaskLog(taskId, logFile);
+
+    verifyAll();
+  }
+
+  @Test
+  public void test_pushTaskReports_shouldUseReportTaskLogs() throws IOException
+  {
+    String taskId = "test-task-id";
+    File logFile = new File("test.log");
+
+    reportTaskLogs.pushTaskReports(taskId, logFile);
+    EasyMock.expectLastCall();
+    replayAll();
+
+    taskLogs.pushTaskReports(taskId, logFile);
+
+    verifyAll();
+  }
+
+  @Test
+  public void test_pushTaskStatus_shouldUseReportTaskLogs() throws IOException
+  {
+    String taskId = "test-task-id";
+    File logFile = new File("test.log");
+
+    reportTaskLogs.pushTaskStatus(taskId, logFile);
+    EasyMock.expectLastCall();
+    replayAll();
+
+    taskLogs.pushTaskStatus(taskId, logFile);
+
+    verifyAll();
+  }
+
+  @Test
+  public void test_pushTaskPayload_shouldUseReportTaskLogs() throws IOException
+  {
+    String taskId = "test-task-id";
+    File logFile = new File("test.log");
+
+    reportTaskLogs.pushTaskPayload(taskId, logFile);
+    EasyMock.expectLastCall();
+    replayAll();
+
+    taskLogs.pushTaskPayload(taskId, logFile);
+
+    verifyAll();
+  }
+
+  @Test
+  public void test_killAll_shouldUseReportTaskLogs() throws IOException
+  {
+    reportTaskLogs.killAll();
+    EasyMock.expectLastCall();
+    replayAll();
+
+    taskLogs.killAll();
+
+    verifyAll();
+  }
+
+  @Test
+  public void test_killOlderThan_shouldUseReportTaskLogs() throws IOException
```

Comment:
Nit: `uses` instead of `shouldUse` reads better.
```suggestion
  public void test_killOlderThan_usesReportTaskLogs() throws IOException
```